crates/turbo-tasks-memory/src/output.rs

```diff
@@ -60,32 +60,8 @@ impl Output {
     }
 
     pub fn link(&mut self, target: RawVc, turbo_tasks: &dyn TurboTasksBackendApi) {
-        let change;
-        let mut _type_change = false;
-        match &self.content {
-            OutputContent::Link(old_target) => {
-                if match (old_target, &target) {
-                    (RawVc::TaskOutput(old_task), RawVc::TaskOutput(new_task)) => {
-                        old_task == new_task
-                    }
-                    (
-                        RawVc::TaskCell(old_task, old_index),
-                        RawVc::TaskCell(new_task, new_index),
-                    ) => old_task == new_task && *old_index == *new_index,
-                    _ => false,
-                } {
-                    change = None;
-                } else {
-                    change = Some(target);
-                }
-            }
-            OutputContent::Empty | OutputContent::Error(_) | OutputContent::Panic(_) => {
-                change = Some(target);
-            }
-        };
-        if let Some(target) = change {
-            self.assign(OutputContent::Link(target), turbo_tasks)
-        }
```

Comment:
What does this fix or change?

Reply:
This does 2 things:
1. Simplifies the equality
2. Removes the equality check from the `link` method

The simplification is just that the old code manually implemented equality, and it wasn't needed. It's much easier to just defer to the equality of the `RawVc` inside the `OutputContent::Link`.

The split out of the `link` method is necessary because I need to check if any invalidation happens in `crates/turbo-tasks-memory/src/task.rs` so that I can log while I still have access to the `Task`.